daily_coding_problem_16_20.py

```diff
@@ -85,39 +85,99 @@ def coding_problem_17(path_str):
     The name of a directory or sub-directory will not contain a period.
     Example:
 
+    >>> coding_problem_17('dir\n\tfile1.ext')
+    13
     >>> coding_problem_17('dir\n\tsubdir1\n\t\tfile1.ext\n\t\tsubsubdir1\n\tsubdir2\n\t\tsubsubdir2\n\t\t\tfile2.ext')
     32
+    >>> coding_problem_17('dir\n\tsubdir1\n\t\tfile1.ext\n\t\tsubsubdir1\n\tsubdir2\n\t\tsubsubdir2\n\t\t\tfile2.ext\ndir2\n\tsubdir1\n\tsubdir2\n\t\tsubsubdir1\n\t\t\tsubsubsubdir3\n\t\t\t\tfile3.ext')
+    47
     """
-    if not path_str:
-        return 0
-
-    max_len = 0
-    dirs = [None]
-    tokens = path_str.split('\n')
-    for token in tokens:
-
-        tabs = 0
-        while token[tabs] == '\t':
-            tabs += 1
-
-        if tabs == len(dirs) - 1:  # substitute current leaf
-
-            dirs[-1] = token
-
-        elif tabs == len(dirs):  # go deeper
-
-            dirs.append(token)
-            max_len = max(max_len, len(' '.join(map(str.strip, dirs))))
-
-        elif tabs > len(dirs):  # malformed string
-
-            raise RuntimeError('Malformed path string')
-
-        else:  # tabs < level
-
-            dirs = dirs[:len(dirs)]
-
-    return max_len
+    class Node(object):
+        '''
+        Node of a Directory N-ary Tree
+        '''
+        def __init__(self, val=None, level=None, parent=None, is_file=False):
+            self.val = val
+            self.level = level
+            self.parent = parent
+            self.subdir = list()    # Node's children as a list
+            self.is_file =is_file
+
+        def add_node(self, node=None):
```

Comment:
I don't think either the add_node or get_parent methods add any semantic value.
What's wrong for example with using node.parent? (see https://goo.gl/KcB4Sk)

Reply:
Yep, I agree. I should have given it a second thought and make it neat. Thanks for pointing it out. ;) And definitely I want to align with conventional coding style as much as possible. Cheers.